pkg/client/request.go

```diff
@@ -540,19 +535,35 @@ func (r *Request) Do() Result {
 				}
 			}
 		}
-		return Result{respBody, created, err, r.codec}
+		body, err := ioutil.ReadAll(r.resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		defer r.resp.Body.Close()
```

Comment:
This defer close should come right after checking the error from the client.Do, I think-- line 523?